ui/src/app/components/apps-panel/expanded-app.spec.tsx

```diff
@@ -0,0 +1,290 @@
+import * as React from 'react';
+import { mount } from 'enzyme';
+
+import {
+  AppsApi,
+  AppStatus,
+  AppType,
+  NotebooksApi,
+  RuntimeStatus,
+  UserAppEnvironment,
+} from 'generated/fetch';
+
+import { environment } from 'environments/environment';
+import {
+  leoAppsApi,
+  leoRuntimesApi,
+} from 'app/services/notebooks-swagger-fetch-clients';
+import { registerApiClient as leoRegisterApiClient } from 'app/services/notebooks-swagger-fetch-clients';
+import { appsApi, registerApiClient } from 'app/services/swagger-fetch-clients';
+import { runtimeStore, serverConfigStore } from 'app/utils/stores';
+import { AppsApi as LeoAppsApi, RuntimesApi } from 'notebooks-generated/fetch';
+
+import defaultServerConfig from 'testing/default-server-config';
+import { waitOneTickAndUpdate } from 'testing/react-test-helpers';
+import { AppsApiStub } from 'testing/stubs/apps-api-stub';
+import { LeoAppsApiStub } from 'testing/stubs/leo-apps-api-stub';
+import { NotebooksApiStub } from 'testing/stubs/notebooks-api-stub';
+import { RuntimeApiStub } from 'testing/stubs/runtime-api-stub';
+import { RuntimesApiStub } from 'testing/stubs/runtimes-api-stub';
+import { workspaceDataStub } from 'testing/stubs/workspaces';
+
+import { ExpandedApp } from './expanded-app';
+import { UIAppType } from './utils';
+
+const googleProject = 'project-for-test';
+const workspace = {
+  ...workspaceDataStub,
+  googleProject,
+};
+const onClickRuntimeConf = jest.fn();
+const onClickDeleteRuntime = jest.fn();
+
+const component = async (
+  appType: UIAppType,
+  initialUserAppInfo: UserAppEnvironment
+) =>
+  mount(
+    <ExpandedApp
+      {...{
+        appType,
+        initialUserAppInfo,
+        workspace,
+        onClickRuntimeConf,
+        onClickDeleteRuntime,
+      }}
+    />
+  );
+
+describe('ExpandedApp', () => {
+  const appsStub = new AppsApiStub();
+  const runtimeStub = new RuntimeApiStub();
+  runtimeStub.runtime.googleProject = googleProject;
+  beforeEach(() => {
+    serverConfigStore.set({ config: defaultServerConfig });
+    registerApiClient(AppsApi, appsStub);
+    registerApiClient(NotebooksApi, new NotebooksApiStub());
+    leoRegisterApiClient(LeoAppsApi, new LeoAppsApiStub());
+    leoRegisterApiClient(RuntimesApi, new RuntimesApiStub());
+    runtimeStore.set({
+      workspaceNamespace: workspace.namespace,
+      runtime: runtimeStub.runtime,
+      runtimeLoaded: true,
+    });
+  });
+
+  test.each([
+    ['should', RuntimeStatus.Running, true],
+    ['should', RuntimeStatus.Stopped, true],
+
+    ['should not', RuntimeStatus.Stopping, false],
+    ['should not', RuntimeStatus.Starting, false],
+    ['should not', RuntimeStatus.Error, false],
+    ['should not', RuntimeStatus.Unknown, false],
+    ['should not', undefined, false],
+    ['should not', null, false],
```

Comment:
Option 3 for test naming: Two separate sets of tests (sharing a helper method that takes a boolean), one for `should` and one for `should not`. This would keep the tests a bit flatter.